determined-ai/determined
# feat: Add Webhook CRUD API [WEB-212]

## Description

This PR introduces a `CRD` (no U here) API and the necessary DB tables and migrations that allow users to create webhooks with defined triggers. Retrieve all webhooks, and delete a webhook. 

Not included in this PR: 
 - RBAC permissions and implementations for the API. If permissible, the plan will be to follow up this PR with another PR that will introduce RBAC auth for the `CRD` actions.  



## Test Plan
1. Navigate to Swagger UI
2. In the POST Webhooks api submit a Webhook payload
For example:

```
{
  "id": 0,
   "url":"http://mywebhookxample.org",
   "triggers":[
      {
         "triggerType":"TRIGGER_TYPE_EXPERIMENT_STATE_CHANGE",
         "condition":{
            "state":"COMPLETED"
         }
      }
   ],
  "webhookType": "WEBHOOK_TYPE_SLACK"
}
```
3. Execute the GET request and ensure any webhooks created are returned.
4. Execute the DELETE request on any defined webhook the request should be succesful, confim that the delete webhook is not in the GET request anymore. 




## Commentary (optional)




## Checklist
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
